### PR TITLE
#163515275 User should be able to rate posts.

### PR DESCRIPTION
**Pull request by espoirMur**
#### What does this PR do?
This PR adds the functionality of creating, updating, fetching rating on a given post

#### Description of Task to be completed?

-  setup all relationship for the database

-  setup Prettier for integration with Eslint

- add create rating endpoint 
- add get all rating endpoint
- add update rating endpoint
- secure all endpoints with Jwt

#### How should this be manually tested?
* Open Postman client
* Consume the API:
    * `post: /api/v1/post/:id/rating` supply the user 's token,and number of rating.
    * `put: /api/v1/post/:postId/rating/:ratingId` supply the user 's token, and the updated rate.
    * `get: api/v1/post/:id/ratingId` supply the user 's token.

#### PS: 

When running all the tests suite , some test fails because of timeout 
If you want to test this feature separately run this command 👍 
`./node_modules/.bin/jest babel-register tests/rating.test.js --forceExit
 --detectOpenHandles`

### CODE REVIEW

File: controllers/ratings.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+class RatingController {
+  static create(req, res, next) {
+    const postId = parseInt(req.params.id, 10);
+    const userId = req.user.id;
+    const { rating } = req.body;
+    joi.validate({ postId, userId, rating }, ratingOne, (err, value) => {
+      if (err) {
+        next(err);
+      } else {
+        Rating.create({ postId, userId, rating })
```
Comment (abayo-luc) [inline controllers/ratings.js:17]: Can you use ```async``` and ```await``` rather than promises?

File: controllers/ratings.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  static getAll(req, res, next) {
+    const postId = parseInt(req.params.id, 10);
+    const userId = req.user.id;
+
+    joi.validate({ userId, postId }, ratingAll, (err, value) => {
+      if (err) {
+        next(err);
+      } else {
+        // Ps: user raw query because using findOne was causing errors because of table aming
+        sequelize
```
Comment (abayo-luc) [inline controllers/ratings.js:49]: What are you using the long way, yet there is Sequelize shorthand for this.

File: controllers/ratings.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    const postId = parseInt(req.params.postId, 10);
+    const ratingId = parseInt(req.params.ratingId, 10);
+    const userId = req.user.id;
+    const { rating } = req.body;
+
+    joi.validate({ postId, userId, rating, ratingId }, ratingOne, (err, value) => {
+      if (err) {
+        next(err);
+      } else {
+        Rating.update({ rating }, { where: { id: ratingId } })
```
Comment (abayo-luc) [inline controllers/ratings.js:75]: I think it would be better if you add the updated rating in the response.

File: controllers/ratings.js (hunk 1 above)
Comment (espoirMur) [reply controllers/ratings.js:17]: I always use promises @abayo-luc , thanks to @Musigwa for teaching me them

File: controllers/ratings.js (hunk 2 above)
Comment (espoirMur) [reply controllers/ratings.js:49]: Check my comment before that, I have tried before but it was not working ...
I decide to go with plain sql

**Pull request merged**